File: engine/engine-model/src/main/scala/server/maps/MapleKite.scala
package server.maps

import java.awt.Point

class MapleKite(private var _ownerName: String, private var _pos: Point, private var _ft: Int, private var _text: String, private var _itemId: Int) extends AbstractMapleMapObject {
  def ownerName: String = _ownerName

  def pos: Point = _pos

  def ft: Int = _ft

  def text: String = _text

  def itemId: Int = _itemId

  override def `type`(): MapleMapObjectType = MapleMapObjectType.KITE

  override def position: Point = pos.getLocation

  override def position_$eq(position: Point): Unit = throw new UnsupportedOperationException
}

File: engine/engine-model/src/main/java/server/maps/MapleKite.java
package server.maps;

import java.awt.Point;

public class MapleKite extends AbstractMapleMapObject {
   private final String ownerName;

   private final Point pos;

   private final Integer ft;

   private final String text;

   private final Integer itemId;

   public MapleKite(String ownerName, Point pos, Integer ft, String text, Integer itemId) {
      this.ownerName = ownerName;
      this.pos = pos;
      this.ft = ft;
      this.text = text;
      this.itemId = itemId;
   }

   public String ownerName() {
      return ownerName;
   }

   public Point pos() {
      return pos;
   }

   public Integer ft() {
      return ft;
   }

   public String text() {
      return text;
   }

   public Integer itemId() {
      return itemId;
   }

   @Override
   public MapleMapObjectType type() {
      return MapleMapObjectType.KITE;
   }

   @Override
   public Point position() {
      return pos.getLocation();
   }

   @Override
   public void setPosition(Point position) {
      throw new UnsupportedOperationException();
   }
}
